internal/provider/resource_rediscloud_database_test.go

```diff
@@ -1,13 +1,138 @@
 package provider
 
 import (
+	"context"
 	"fmt"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
-	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
 	"os"
+	"strconv"
 	"testing"
+
+	"github.com/RedisLabs/rediscloud-go-api/redis"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
 )
 
+// Checks CRUDI (CREATE,READ,UPDATE,IMPORT) operations on the database resource.
+func TestAccResourceRedisCloudDatabase_CRUDI(t *testing.T) {
+
+	name := acctest.RandomWithPrefix(testResourcePrefix)
+	password := acctest.RandString(20)
+	resourceName := "rediscloud_database.example"
+	subscriptionResourceName := "rediscloud_subscription.example"
+	replicaResourceName := "rediscloud_database.example_replica"
+	testCloudAccountName := os.Getenv("AWS_TEST_CLOUD_ACCOUNT_NAME")
+
+	var subId int
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t); testAccAwsPreExistingCloudAccountPreCheck(t) },
+		ProviderFactories: providerFactories,
+		CheckDestroy:      testAccCheckSubscriptionDestroy,
+		Steps: []resource.TestStep{
+			// Test database and replica database creation
+			{
+				Config: fmt.Sprintf(testAccResourceRedisCloudDatabase, testCloudAccountName, name, password) + testAccResourceRedisCloudDatabaseReplica,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", "example"),
+					resource.TestCheckResourceAttr(resourceName, "protocol", "redis"),
+					resource.TestCheckResourceAttr(resourceName, "memory_limit_in_gb", "3"),
```

Comment:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "memory_limit_in_gb", "1"),
```

Reply:
This was set to 3 initially so that it can be updated in the next step to 1 (see the post-update check on line 97)